### ISSUE TITLE
User name should be checked for allowed symbols

### ISSUE THREAD

**Issue by inliquid** [esotalk/esoTalk#311]
Otherwise he may take user name which will break mentions (and of course may cause other errors or even lead to vulnerability). F.i. user with name user@ will not be triggered by regexp of mentions() and getMentions().

I suggest to use mb_regex_encoding()+mb_ereg_replace() and filter out only alphanumeric symbols, spaces, underscores "_" and "-". Other symbols turn into "_".

**Comment by tobyzerner** [comment]
Originally usernames were limited to alphanumeric characters, but this was removed because a lot of forums in different languages required glyphs/other characters in usernames. Also it is possible that forums will want to allow full names in usernames (i.e. containing a space) — I think this specific case IS currently handled in @mentions. 

What would be the happy medium here? Maybe just disallowing @ symbols in usernames but nothing else?

**Comment by inliquid** [comment]
Yes you are right, that somehow this limitation should be connected to native language...

F.i. I have added this code, which allows Russian and alphanumeric characters.

```
            mb_regex_encoding('UTF-8');
    $data["username"] = mb_ereg_replace("[^ёЁА-Яа-яA-Za-z0-9_\- ]", "_", $data["username"]);
```

Maybe take out these characters to config, in order to let administrator assign it?

**Comment by inliquid** [comment]
If you disallow '@', I think you have to disallow other special characters, such as '&', quotes, punctuation, and so on.
